monitoring/ceph-mixin/README.md

```diff
@@ -0,0 +1,40 @@
+## Ceph mixin
+
+The ceph-mixin generates both grafana dashboards and alerts with Jsonnet.
+You can find all the generated files in the `output` directory and you can
+run `tox -egrafonnet-fix` to re-generate them.
+
+The ceph-mixin depends on some Jsonnet third party libraries located in the
+`vendor` directory. To update those libraries you will have to install
+[jsonnet-bundler](https://github.com/jsonnet-bundler/jsonnet-bundler) and run
+the command `jb update`.
```

Comment:
We discussed the versioning of grafonnet files in the original PR. What's the reason not to download them at build time? Is it a best practice in the jsonnet/mixin world? If not, couldn't we instead add them as a `git subtree`?

Reply:
I didn't do it because this is managed by jsonnet-bundler and I thought that it would add another dependency. A `git subtree` would probably be in conflict with what jsonnet is doing :/.

I think we should choose between:
1. Having the final json and yaml for the dashboards and grafana in the repo with grafonnet
2. Having the final json and yaml for the dashboards and grafana in the repo without grafonnet BUT having jsonnet-bundler at least in the CI that check if the final json files are up to date
3. Having neither grafonnet nor the final yaml/json in the repo but add jsonnet and jsonnet-bundler as a build dependency

1) is convenient because you don't have to deal with dependencies
2) is what is done in master today but with a bit more complexity since we would have to install a go software not packaged for our distro (we could probably do a go install somewhere instead if we choose this)
3) need to have the dependencies installed at both build time and check time and is not super convenient for users that just want to download the json/yaml from the repo

I didn't see any project that keeps the vendor directory in their repo (I didn't see any that was directly in the build system and have the need to be packaged though). Some includes the final json/yaml directly in the repo some don't it depends.

Reply:
After thinking about this again, I would say adding go as a builld dependency and build jsonnet-bundler would be worth it because I want to add some other go program for the grafana dashboard (for instance this one https://github.com/grafana/dashboard-linter).

I don't know if we want to do the 3rd option (so about not adding the final json/yaml in the repo but building it). Maybe we want to wait for the missing dashboard which is not generated with jsonnet to be also generated with jsonnet before doing that (`ceph-cluster.json` IIRC).

Reply:
@MrFreezeex thank you! Personally:
* I like having both the input-output files versioned, as that allows us to quickly detect how jsonnet/grafonnet changes impact on the generated JSONs.
* I'd prefer not to have the grafonnet code versioned, but if necessary, the `git tree`/`submodules` would be my preferred choices... :-/
* Regarding the build deps, sooner or later (hopefully sooner) we should end up building Ceph in a containerized pipeline, so we will be able to run some pipeline stages in the specific environments where the required tools are available (jsonnet, jsonnet-bundler, promtool, etc).

That said, I'm all for a pragmatic approach and you seem to be quite experienced in this stuff, so I'm ok with it! I'd also like to hear from other Ceph devels that reviewed [the original PR](https://github.com/ceph/ceph/pull/42194): @tchaikov @p-se @smithfarm @pcuzner @djgalloway @ktdreyer

Reply:
@epuertat I pushed two commits about that (the wip commit to be squashed ofc):
- https://github.com/ceph/ceph/pull/44059/commits/ad9ac597bf0682adf245305a7072a91aa6317d8b
- https://github.com/ceph/ceph/pull/44059/commits/d6beb370b2b2e8fddee37ae63c8a666af2c211ea

It adds go as a build dependency and install/run jsonnet bundler for make check. WDYT?

Reply:
@MrFreezeex thanks a lot! Makes sense. I don't think installing `go` would be anything quirky in a project closely related to k8s.